src/Orders/OrderScreen.js

```diff
@@ -0,0 +1,180 @@
+            if (err) { console.error(err); return; }
+            order.fields.produceItem = prodRecord;
+            if (!itemsListVar.has(currDate)) {
+              itemsListVar.set(currDate, prodRecord.fields.Name);
+              imagesVar = imagesVar.set(currDate, prodRecord.fields.Image[0].url);
+            } else if (itemsListVar.get(currDate).indexOf(',') === -1) {
+              itemsListVar.set(currDate, `${itemsListVar.get(currDate)}, ${prodRecord.fields.Name}`);
+            } else {
+              const ind = itemsListVar.get(currDate).indexOf('+');
+              if (ind === -1) {
+                itemsListVar = itemsListVar.set(currDate, `${itemsListVar.get(currDate)} + 1 more`);
+                itemsListCount = itemsListCount.set(currDate, 1);
+              } else {
+                itemsListCount = itemsListCount.set(currDate, itemsListCount.get(currDate) + 1);
+                let newStr = itemsListVar.get(currDate);
+                newStr = newStr.substring(0, itemsListVar.get(currDate).indexOf('+') - 1);
+                newStr += ` + ${itemsListCount.get(currDate)} more`;
+                itemsListVar = itemsListVar.set(currDate, newStr);
+              }
+            }
+            setItemsList(itemsListVar);
+            setImages(imagesVar);
+            setFlag((f) => !f);
+          }));
+        } else {
+          console.log('ERR: misssing produce id in orders record');
+        }
+
+        if (!monthCondenseOrders.has(currMonthYear)) {
+          monthCondenseOrders.set(currMonthYear, new Map([[currDate, [order.fields]]]));
+        } else if(!monthCondenseOrders.get(currMonthYear).has(currDate)) {
+          monthCondenseOrders.get(currMonthYear).set(currDate, [order.fields]);
+        } else {
+          monthCondenseOrders.get(currMonthYear).get(currDate).push(order.fields);
+        }
+      });
+      setOrderMap(new Map([...monthCondenseOrders].sort((a, b) => new Date(a[0]) - new Date(b[0]))));
+      fetchNextPage();
+    }, (err) => {
+      if (err) {
+        Alert.alert(err.message);
+      }
+    });
+  };
+
+  useEffect(() => {
+    getOrders();
+  }, []);
+
+  useEffect(() => {
+    let i = 0;
+    const OrderCards = (Array.from(orderMap)).map((items) => {
+    return (
+      <>
+      <Title style={styles.subtitleText}> {MONTHS[new Date(items[0]).getMonth()]} Orders </Title>
+      <View style={styles.orderCardsContainer}>
+        {
+        (Array.from(new Map([...items[1]].sort((a, b) => new Date(a[0]) - new Date(b[0]))))).map((card) => {
+          return (
+          <OrderCard
+            style={styles.orderCard}
+            navigation={navigation}
+            orderId={(i++).toString()}
+            key={(i).toString()}
+            items={card}
+            itemsList={itemsList}
+            images={images}
+          />
+          )
+        })
+        }
+      </View>
+      </>
+      )
+    });
+    setCardList(OrderCards);
+  }, [itemsList, flag]);
```

Comment:
Could you leave a comment in the code about why there's two useEffects? The first is forthe initial render, that much I get intuitively... What is the second useEffect doing?

Reply:
added a comment.
 the second useeffect, just updates the ordercards once the itemList variables get updated after the airtable api calls succeed. This is necessary because these api calls take a ton of time, and the rendering needs to be refreshed after they finish.